test/src/unit-capi-array.cc

```diff
@@ -1484,10 +1486,13 @@ TEST_CASE_METHOD(
   rc = tiledb_query_set_data_buffer(
       ctx_, query, "a", buffer_a1, &buffer_a1_size);
   CHECK(rc == TILEDB_OK);
-  rc = tiledb_query_submit(ctx_, query);
-  CHECK(rc == TILEDB_OK);
-  rc = tiledb_query_finalize(ctx_, query);
-  CHECK(rc == TILEDB_OK);
+  // rc = tiledb_query_submit(ctx_, query);
+  // CHECK(rc == TILEDB_OK);
+  // rc = tiledb_query_finalize(ctx_, query);
+  // CHECK(rc == TILEDB_OK);
+  // TODO: Bug, wrong results
+  submit_serialized_query(ctx_, query);
```

Comment:
In create_dense_vector, set the capacity to 10 on the array schema. Should fix this case.